### ISSUE TITLE
NTLM scenario is broken after recent updates

### ISSUE THREAD

**Issue by koltyakov** [s-KaiNet/node-sp-auth#115]
Looks that NTLM auth has been broken after some recent updates. Please check: https://github.com/koltyakov/sp-rest-proxy/issues/151

It's probably related to `got` library.

It's reproducible on any bare minimal code setup, not only in sp-rest-proxy.

```bash
=== Error === 
 TypeError: Expected the `options.agent` properties to be `http`, `https` or `http2`, got `_events`
    at normalizeArguments (/Users/andrew/Projects/GitHub/node-sp-auth-troubleshoot/node_modules/got/dist/source/core/index.js:685:27)
    at Object.got [as request] (/Users/andrew/Projects/GitHub/node-sp-auth-troubleshoot/node_modules/got/dist/source/create.js:112:39)
    at OnpremiseUserCredentials.getAuth (/Users/andrew/Projects/GitHub/node-sp-auth-troubleshoot/node_modules/node-sp-auth/lib/src/auth/resolvers/OnpremiseUserCredentials.js:29:25)
    at getAuth (/Users/andrew/Projects/GitHub/node-sp-auth-troubleshoot/node_modules/node-sp-auth/lib/src/index.js:16:76)
    at /Users/andrew/Projects/GitHub/node-sp-auth-troubleshoot/index.js:22:30
```

**Comment by s-KaiNet** [comment]
Published reverted version 3.0.7. Please check if it works fine now.

**Comment by koltyakov** [comment]
It works like a charm! Thank you, Sergei!
